StoreMessageContentActivity/handler.ts

```diff
@@ -53,13 +67,102 @@ export type StoreMessageContentActivityResult = t.TypeOf<
   typeof StoreMessageContentActivityResult
 >;
 
+export const ServicePreferenceError = t.interface({
+  kind: t.keyof({ ERROR: null, LEGACY: null }),
+  message: t.string
+});
+
+export type ServicePreferenceError = t.TypeOf<typeof ServicePreferenceError>;
+
+export type ServicePreferenceValueOrError = (params: {
+  readonly serviceId: NonEmptyString;
+  readonly fiscalCode: FiscalCode;
+  readonly userServicePreferencesMode: ServicesPreferencesMode;
+  readonly userServicePreferencesVersion: number;
+}) => TaskEither<
+  ServicePreferenceError,
+  ReadonlyArray<BlockedInboxOrChannelEnum>
+>;
+
+const servicePreferenceToBlockedInboxOrChannels: (
+  servicePreference: ServicePreference
+) => ReadonlyArray<BlockedInboxOrChannelEnum> = servicePreference => {
+  const blockedInboxOrChannels = [];
+  if (!servicePreference.isInboxEnabled) {
+    // eslint-disable-next-line functional/immutable-data
+    blockedInboxOrChannels.push(BlockedInboxOrChannelEnum.INBOX);
+  }
+  if (!servicePreference.isEmailEnabled) {
+    // eslint-disable-next-line functional/immutable-data
+    blockedInboxOrChannels.push(BlockedInboxOrChannelEnum.EMAIL);
+  }
+  if (!servicePreference.isWebhookEnabled) {
+    // eslint-disable-next-line functional/immutable-data
+    blockedInboxOrChannels.push(BlockedInboxOrChannelEnum.WEBHOOK);
+  }
+  return blockedInboxOrChannels;
+};
+
```

Comment:
To avoid `if` and non-immutable values. This also prevents us to forget to handle a new channel: map `channelToBlockedInboxOrChannelEnum` it's going to break the build in that case.

```suggestion
type ServicePreferencesValues = Omit<
  ServicePreference,
  "serviceId" | "fiscalCode" | "settingsVersion"
>;
const channelToBlockedInboxOrChannelEnum: {
  readonly [key in keyof ServicePreferencesValues]: BlockedInboxOrChannelEnum;
} = {
  isInboxEnabled: BlockedInboxOrChannelEnum.INBOX,
  isEmailEnabled: BlockedInboxOrChannelEnum.EMAIL,
  isWebhookEnabled: BlockedInboxOrChannelEnum.WEBHOOK
};

const servicePreferenceToBlockedInboxOrChannels: (
  servicePreference: ServicePreference
) => ReadonlyArray<BlockedInboxOrChannelEnum> = servicePreference =>
  Object.entries(servicePreference)
    // take only attributes of ServicePreferencesValues
    .filter(([name, _]) => channelToBlockedInboxOrChannelEnum[name])
    // take values set to false
    .filter(([_, isEnabled]) => !isEnabled)
    // map to BlockedInboxOrChannelEnum
    .map(([name, _]) => channelToBlockedInboxOrChannelEnum[name]);
```

Reply:
Applied with #af4c670. I think this code is very smart.